repaygithub/cactus
# CACTUS-745 :: StatusMessage Updates

Ticket: https://repayonline.atlassian.net/browse/CACTUS-745

I did a bit of refactoring so you may want to review commits individually. Mainly the refactoring was to clean up the component and use some of the newer style helpers, but there's also some refactoring I had to do to ignore the `info` status for other components.

We were previously importing the `Status` type from the `StatusMessage` file for components that also support those props, but since those components don't support the new `info` status, I moved the type definition to `helpers/status` and then extended from that type to include `info` for `StatusMessage`.

### Testing
1. Open up the new `StatusMessage` story and check out the info variant to make sure it looks right.
2. Inspect the page and set `color: white` div containing the `StatusMessage`s. Make sure the text inside the messages don't reflect that white foreground color.
